### ISSUE TITLE
Same parameter in path and requestBody

### ISSUE THREAD

**Issue by m-develops** [schemathesis/schemathesis#1048]
Is it possible to make sure the same generated path parameter also comes back in an attribute in the requestBody before this is send to the server.
This is for example required for JSON:API requests:
https://jsonapi.org/format/#crud-updating

Could this be possible with hooks for example?

I have setup links so the id from the response of the POST request is in the parameter of the PATCH request, but i can not seem to get the id in some part of the requestBody changed.

For example i tried the example below but the requestBody is not changed
```
    UpdateDocument:
      operationId: updateDocument
      parameters:
        id: '$response.body#/data/id'
      requestBody: '$response.body'
```

Schemathesis command:
```
	schemathesis run --junit-xml=junit.xml --stateful=links --hypothesis-suppress-health-check=too_slow
	--hypothesis-max-examples=3 --store-network-log=schemathesis-log.yaml --checks all --verbosity \
	--hypothesis-verbosity=quiet --hypothesis-deadline=None --hypothesis-phases=reuse,generate,target,shrink \
	--base-url=http://localhost/api dist/document-api.yaml
```

**Comment by Stranger6667** [comment]
Hi @m-develops 

Thank you for reporting this, and sorry for the late reply. Unfortunately, there is no direct way to do so in CLI at the moment.

I believe that it could be done via a hook planned in #700 or similar. Probably from the Hypothesis point of view, it would be better to have a hook that will allow the user to add a `map` to the strategy. May be some `before_generate_case` hook similar to other `before_generate_*` hooks:

```python
@schemathesis.hooks.register
def before_generate_case(context, strategy):

    def tune_case(case):
        if context.operation.method == "PATCH" and context.operation.path == "/users/":
            case.path_parameters["id"] = case.body["data"]["id"]  # Or vice-versa
        return case

    return strategy.map(tune_case)
```

I'll work on this hook soon.

Re: links. It is a bug :( I forgot to process the collected `body` variants; now, only non-body parameters are processed in CLI. Additionally, `'$response.body'` is parsed into a string, which is wrong and should preserve the original body type - otherwise `"{}"` will become `"\"{}\""`.  I'll create separate issues for these problems.

**Comment by Stranger6667** [comment]
All the mentioned changes are implemented in Schemathesis `3.5.0`. The hook from my previous comment can be attached like this:

```
schemathesis --pre-run hooks run ... # other CLI options & arguments
```

Assuming `hooks.py` available in the directory where `schemathesis` is invoked.
Feel free to open a new issue if those changes aren't sufficient for your use case.
